snapshots/windows/windows.go

```diff
@@ -312,12 +314,19 @@ func (s *snapshotter) mounts(sn storage.Snapshot) []mount.Mount {
 		roFlag = "rw"
 	}
 
-	if len(sn.ParentIDs) == 0 || sn.Kind == snapshots.KindActive {
-		source = s.getSnapshotDir(sn.ID)
-		parentLayerPaths = s.parentIDsToParentPaths(sn.ParentIDs)
-	} else {
-		source = s.getSnapshotDir(sn.ParentIDs[0])
-		parentLayerPaths = s.parentIDsToParentPaths(sn.ParentIDs[1:])
+	source := s.getSnapshotDir(sn.ID)
+	parentLayerPaths := s.parentIDsToParentPaths(sn.ParentIDs)
+
+	mountType := "windows-layer"
+
+	if len(sn.ParentIDs) == 0 {
+		// A mount of a parentless snapshot is a bind-mount.
+		mountType = "bind"
+		// If not being extracted into, then the bind-target is the
+		// "Files" subdirectory.
+		if !strings.Contains(key, snapshots.UnpackKeyPrefix) {
+			source = filepath.Join(source, "Files")
+		}
```

Comment:
Maybe this logic (appending `Files`) should actually live in the mount code?

That would make this key-check unnecessary, as the assumption underlying the current code's correctness is that imports never mount the directory, and the only users of the Source key are importing and mounting. It'd be nice to remove that assumption, since for example I've just been reminded we pass the `Mount` object out into the runtime shim, so avoiding surprises there seems valuable.

That'd also mean one fewer places that rely on the snapshot _key name_, which I'm not super-fond of anyway.

Reply:
Moved this append in the ```mount()``` function. Running tests now. If all is well, I'll update the PR. Good catch!

Reply:
Seems ok.